src/hipscat/pixel_math/box_filter.py

```diff
@@ -0,0 +1,146 @@
+        result_pixels = get_filtered_pixel_list(pixel_tree, filter_tree)
+
+    return result_pixels
+
+
+def wrap_ra_values(ra: Tuple[float, float] | None) -> Tuple[float, float] | None:
+    """Wraps right ascension values to the [0,360] degree range and sorts
+    declination values by ascending order.
+
+    Args:
+        ra (Tuple[float, float]): The right ascension values, in degrees
+
+    Returns:
+        The right ascension values wrapped to the [0,360] degree range,
+        or None if they do not exist.
+    """
+    return tuple(wrap_angles(list(ra))) if ra else None
+
+
+def wrap_angles(ra: List[float]) -> List[float]:
+    """Wraps angles to the [0,360] degree range.
+
+    Arguments:
+        ra (List[float]): List of right ascension values
+
+    Returns:
+        A list of right ascension values, wrapped to the [0,360] degree range.
+    """
+    return Angle(ra, u.deg).wrap_at(360 * u.deg).degree
+
+
+def _generate_ra_strip_pixel_tree(ra_range: Tuple[float, float], order: int) -> PixelTree:
+    """Generates a pixel_tree filled with leaf nodes at a given order that overlap with the ra region."""
+    # Subdivide polygons (if needed) in two smaller polygons of at most 180 degrees
+    division_ra = _get_division_ra(ra_range)
+
+    if division_ra is not None:
+        pixels_in_range = _get_pixels_for_subpolygons(
+            [
+                # North Pole
+                [(0, 90), (ra_range[0], 0), (division_ra, 0)],
+                [(0, 90), (division_ra, 0), (ra_range[1], 0)],
+                # South Pole
+                [(ra_range[0], 0), (0, -90), (division_ra, 0)],
+                [(division_ra, 0), (0, -90), (ra_range[1], 0)],
+            ],
+            order,
+        )
+    else:
+        # HEALpy will assume the polygon of smallest area by default
+        # e.g. for ra_ranges of [10,50], [200,10] or [350,10]
+        pixels_in_range = _get_pixels_for_subpolygons(
+            [
+                # North Pole
+                [(0, 90), (ra_range[0], 0), (ra_range[1], 0)],
+                # South Pole
+                [(ra_range[0], 0), (0, -90), (ra_range[1], 0)],
+            ],
+            order,
+        )
+
+    pixel_list = [HealpixPixel(order, polygon_pixel) for polygon_pixel in pixels_in_range]
+    return PixelTreeBuilder.from_healpix(pixel_list)
+
+
+def _generate_dec_strip_pixel_tree(dec_range: Tuple[float, float], order: int) -> PixelTree:
+    """Generates a pixel_tree filled with leaf nodes at a given order that overlap with the dec region."""
+    nside = hp.order2nside(order)
+    # Convert declination values to colatitudes, in radians, and revert their order
+    colat_rad = np.sort(np.radians([90 - dec if dec > 0 else 90 + abs(dec) for dec in dec_range]))
+    # Figure out which pixels in nested order are contained in the declination band
+    pixels_in_range = hp.ring2nest(
+        nside, hp.query_strip(nside, theta1=colat_rad[0], theta2=colat_rad[1], inclusive=True)
```

Comment:
I think `query_strip` has a `nest=True` parameter that would mean you don't have to do the `ring2nest` call

Reply:
This one is rather interesting! I attempted to use `query_strip` with the `nest` parameter but for a reason that exceeds my knowledge of healpy, I am running into what looks like a segmentation fault. The solution was to wrap the `query_strip` call with a call to `ring2nest`.

Looking it up online I found that the [issue](https://github.com/healpy/healpy/issues/843) about adding the nest flag to query_strip is still open. I don't know what to make of it at the moment so I just might open an issue on healpy to report what I am experiencing!

Reply:
Huh, fair enough. I mean if they already have an issue for the same thing there's no need to make a new one. I think you may have even bumped the old one by mentioning it here. But yeah this work around is fine for now.